### PR TITLE
[Log] Changes to epoch hash.

### PR DESCRIPTION
**Pull request by d0cd**
This PR logs changes to the `current_epoch_hash` in the `Ledger`.

### CODE REVIEW

File: ledger/src/advance.rs [hunk 1]
```diff
@@ -104,6 +104,19 @@ impl<N: Network, C: ConsensusStorage<N>> Ledger<N, C> {
...
+            match epoch_hash {
+                Some(epoch_hash) => {
+                    info!("Updating the `current_epoch_hash` at block height {height} to {epoch_hash}");
+                }
+                None => {
+                    info!("Failed to update the `current_epoch_hash` at block height {height}");
+                }
+            }
             // Update the current epoch hash.
             self.current_epoch_hash.write().clone_from(&self.get_epoch_hash(block.height()).ok());
```
Comment (raychu86) [inline ledger/src/advance.rs]: You already derived the epoch_hash above. `self.get_epoch_hash(block.height()` is an extra read.

File: ledger/src/advance.rs [hunk 2]
```diff
@@ -104,6 +104,19 @@ impl<N: Network, C: ConsensusStorage<N>> Ledger<N, C> {
...
+            let height = block.height();
+            // Get the epoch hash for the block.
+            let epoch_hash = &self.get_epoch_hash(height).ok();
+            // Log the update of the current epoch hash.
+            match epoch_hash {
+                Some(epoch_hash) => {
+                    info!("Updating the `current_epoch_hash` at block height {height} to {epoch_hash}");
+                }
+                None => {
+                    info!("Failed to update the `current_epoch_hash` at block height {height}");
```
Comment (raychu86) [inline ledger/src/advance.rs]: This should not be an `info` log

Comment (howardwu) [reply ledger/src/advance.rs]: `info!` should be reserved only for the most important user-relevant data. This is not relevant data to the user.

Comment (howardwu) [inline ledger/src/advance.rs]:
```suggestion
                    error!("Failed to update the current epoch hash at block {height}");
```

File: ledger/src/advance.rs [hunk 3]
```diff
@@ -104,6 +104,19 @@ impl<N: Network, C: ConsensusStorage<N>> Ledger<N, C> {
...
         // If the block is the start of a new epoch, or the epoch hash has not been set, update the current epoch hash.
         if block.height() % N::NUM_BLOCKS_PER_EPOCH == 0 || self.current_epoch_hash.read().is_none() {
+            // Get the height of the block.
+            let height = block.height();
+            // Get the epoch hash for the block.
+            let epoch_hash = &self.get_epoch_hash(height).ok();
+            // Log the update of the current epoch hash.
+            match epoch_hash {
+                Some(epoch_hash) => {
+                    info!("Updating the `current_epoch_hash` at block height {height} to {epoch_hash}");
```
Comment (howardwu) [inline ledger/src/advance.rs]:
```suggestion
                    trace!("Updating the current epoch hash at block {height} to '{epoch_hash}'");
```

File: ledger/src/advance.rs (hunk 1 above)
Comment (d0cd) [reply ledger/src/advance.rs]: Good catch!

**Comment by howardwu** [comment]
^CI is failing

File: ledger/src/advance.rs [hunk 4]
```diff
@@ -104,8 +104,21 @@ impl<N: Network, C: ConsensusStorage<N>> Ledger<N, C> {
 
         // If the block is the start of a new epoch, or the epoch hash has not been set, update the current epoch hash.
         if block.height() % N::NUM_BLOCKS_PER_EPOCH == 0 || self.current_epoch_hash.read().is_none() {
+            // Get the height of the block.
+            let height = block.height();
+            // Get the epoch hash for the block.
+            let epoch_hash = &self.get_epoch_hash(height).ok();
```
Comment (howardwu) [inline ledger/src/advance.rs]: Can you one-line this?

File: ledger/src/advance.rs [hunk 5]
```diff
@@ -104,8 +104,21 @@ impl<N: Network, C: ConsensusStorage<N>> Ledger<N, C> {
...
+                Some(epoch_hash) => {
+                    trace!("Updating the current epoch hash at block {height} to '{epoch_hash}'");
+                }
+                None => {
+                    error!("Failed to update the current epoch hash at block {height}");
+                }
+            }
             // Update the current epoch hash.
-            self.current_epoch_hash.write().clone_from(&self.get_epoch_hash(block.height()).ok());
+            self.current_epoch_hash.write().clone_from(epoch_hash);
```
Comment (howardwu) [inline ledger/src/advance.rs]: You can move this into L114

File: ledger/src/advance.rs [hunk 6]
```diff
@@ -104,8 +104,21 @@ impl<N: Network, C: ConsensusStorage<N>> Ledger<N, C> {
 
         // If the block is the start of a new epoch, or the epoch hash has not been set, update the current epoch hash.
         if block.height() % N::NUM_BLOCKS_PER_EPOCH == 0 || self.current_epoch_hash.read().is_none() {
+            // Get the height of the block.
+            let height = block.height();
+            // Get the epoch hash for the block.
+            let epoch_hash = &self.get_epoch_hash(height).ok();
+            // Log the update of the current epoch hash.
+            match epoch_hash {
```
Comment (howardwu) [inline ledger/src/advance.rs]: You can move the one-line to this match statement

**Review by howardwu — APPROVED** [review]

**Pull request merged**